sdk/identity/azure-identity/azure/identity/_internal/msal_credentials.py

```diff
@@ -116,3 +186,115 @@ def _get_app(self):
         if not self._msal_app:
             self._msal_app = self._create_app(msal.PublicClientApplication)
         return self._msal_app
+
+
+class InteractiveCredential(PublicClientCredential):
+    def __init__(self, **kwargs):
+        self._disable_automatic_authentication = kwargs.pop("disable_automatic_authentication", False)
+        self._auth_record = kwargs.pop("authentication_record", None)  # type: Optional[AuthenticationRecord]
```

Comment:
(I don't have an answer) Which class is the best to store _auth_record? InteractiveCredential? PublicClientCredential? MsalCredential?

Reply:
This one. It and its subclasses are the only classes which handle `AuthenticationRecord`.